### PR TITLE
Compute: Customization support for pipeline view p1

### PR DESCRIPTION
**Pull request by andrey-sharapov-sc**
- allow function views to accept DG.FuncCall
- reimplement onFuncCallReady as a method
- use funcCallReplaced as a single before draw event API
- proper filtering of step calls via ids
- add some expiremental hooks in pipeline view

### CODE REVIEW

File: libraries/compute-utils/function-views/src/function-view.ts [hunk 1]
```diff
@@ -14,46 +14,46 @@ export abstract class FunctionView extends DG.ViewBase {
...
    * @param options Configuration object for the view.
    */
   constructor(
-    protected funcName: string,
+    protected initValue: string | DG.FuncCall,
     public options: {historyEnabled: boolean, isTabbed: boolean} = {historyEnabled: true, isTabbed: false},
   ) {
     super();
     this.box = true;
+    if (initValue instanceof DG.FuncCall) {
```
Comment (skalkin) [inline libraries/compute-utils/function-views/src/function-view.ts:35]: Let's shorten it:

`this.funcName = initValue instanceof DG.FuncCall ? initValue.func.name : initValue;`

Comment (AufarZakiev) [reply libraries/compute-utils/function-views/src/function-view.ts]: Even better, we can safely remove these lines. `funcName` is never used in the library and never used in third-paty code (just checked).

File: libraries/compute-utils/function-views/src/function-view.ts [hunk 2]
```diff
@@ -192,13 +192,20 @@ export abstract class FunctionView extends DG.ViewBase {
    * @stability Stable
   */
   protected async loadFuncCallById() {
+    if (this.initValue instanceof DG.FuncCall) {
+      // next tick is needed to run funcCallReplaced before building UI
+      await new Promise(resolve => setTimeout(resolve, 0));
```
Comment (AufarZakiev) [inline libraries/compute-utils/function-views/src/function-view.ts:193]: I can't get point of this line. 
`this.funcCallReplaced` only has meaning when UI is already built. Using this event, we can update UI when `this.funcCall` is replaced. By the moment of this line, UI is not built yet. 

Tried to comment this line and run my code - it works properly.

File: libraries/compute-utils/function-views/src/pipeline-view.ts [hunk 3]
```diff
@@ -75,7 +75,14 @@ export class PipelineView extends ComputationView {
 
     this.subs.push(
       grok.functions.onAfterRunAction.pipe(
-        filter((run) => Object.keys(this.steps).includes(run.func.nqName)),
+        filter((run) => {
+          for (const {view} of Object.values(this.steps)) {
```
[the remark this answers is not in the document]
Comment (AufarZakiev) [reply libraries/compute-utils/function-views/src/pipeline-view.ts]: Could be replaced by 
```
filter((run) => Object.values(this.steps).some(({view}) => (view?.funcCall?.id === run?.id) && run)),
```

File: libraries/compute-utils/function-views/src/function-view.ts (hunk 2 above)
Comment (AufarZakiev) [reply libraries/compute-utils/function-views/src/function-view.ts:193]: Ah, got it

**Review by AufarZakiev — APPROVED** [review]

**Pull request merged**